src/Uno.Extensions.Reactive.Generator/Utils/RoslynExtensions.cs

```diff
@@ -109,4 +112,28 @@ public static string ToCSharpCodeString(this Accessibility accessibility)
 
 		throw new ArgumentOutOfRangeException($"{accessibility} is not supported.");
 	}
+
+	public static bool IsPartial(this INamedTypeSymbol type)
+	{
+		var syntaxRefs = type.DeclaringSyntaxReferences;
+		var isPartial = syntaxRefs.Length switch
+		{
+			0 => true,
```

Comment:
what is the reasoning behind the 0 case?

Reply:
Depending of when your generator runs, you can have some code that is not compiling and roslyn was able to determine that it should be a class type, but was not able yet where the class was defined. It should not occur in our case, but it's always better handle all cases!